### PR TITLE
refactor(standardized form data): refine interface and improve code smells

### PR DESCRIPTION
**Pull request by zhaoyongjie**
### SUMMARY

This PR intends to clean up code and refine interfaces. there are no functional changes.
1. rename `denormalizeFormData` to `formDataOverrides` in ControlPanelConfig
2. a simpler approach for transformation.
3. remove `updateStandardizedState` in ControlPanelConfig
4. remove updateStandardizedState callback from each plugins
5. add `size`, `entity`, and `series` to `StandardizedControls`

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

N/A

### TESTING INSTRUCTIONS

CI

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by kgabryje** [comment]
I think I found a bug when moving between Line Chart, Big Number and Pie chart. When I add 2 metrics while on Line Chart and then I switch to Big Number, it uses the first metric(correctly). Then I replace a metric on Big Number and switch to Pie chart. Pie Chart shows the second metric from Line chart instead of the metric from Big Number

https://user-images.githubusercontent.com/15073128/176149136-edf572d3-d952-496a-9b2c-fe43ec73cce5.mov

File: superset-frontend/packages/superset-ui-core/src/query/types/QueryFormData.ts [hunk 1]
```diff
@@ -220,8 +220,4 @@ export function isDruidFormData(
   return 'granularity' in formData;
 }
 
-export function isSavedMetric(metric: QueryFormMetric): metric is SavedMetric {
```
Comment (zhaoyongjie) [inline superset-frontend/packages/superset-ui-core/src/query/types/QueryFormData.ts:223]: move to `query/types/Metric.ts`

**Review by kgabryje — APPROVED** [review]
Tested, works great!

**Pull request merged**